Resources/doc/first_steps.rst

```diff
@@ -249,6 +249,226 @@ to easily fetch objects based on multiple conditions:
+`ExpressionLanguage component`_:
+
+.. code-block:: php
+
+    #[Route('/product/{product_id}')]
+    public function show(
+        #[MapDocument(expr: 'repository.find(product_id)')]
+        Product $product
+    ): Response {
+    }
+
+In the expression, the ``repository`` variable will be your document's
+Repository class and any route wildcards - like ``{product_id}`` are
+available as variables.
+
+This can also be used to help resolve multiple arguments:
+
+.. code-block:: php
+
+    #[Route('/product/{id}/comments/{comment_id}')]
+    public function show(
+        Product $product,
+        #[MapDocument(expr: 'repository.find(comment_id)')]
+        Comment $comment
+    ): Response {
+    }
+
+In the example above, the ``$product`` argument is handled automatically,
+but ``$comment`` is configured with the attribute since they cannot both follow
+the default convention.
+
+MapDocument Options
+~~~~~~~~~~~~~~~~~~~
+
+A number of options are available on the ``MapDocument`` attribute to
+control behavior:
+
+``id``
+    If an ``id`` option is configured and matches a route parameter, then
+    the resolver will find by the primary key:
+
+.. code-block:: php
+
+        #[Route('/product/{product_id}')]
+        public function show(
+            #[MapDocument(id: 'product_id')]
+            Product $product
+        ): Response {
+        }
+
+``mapping``
+    Configures the properties and values to use with the ``findOneBy()``
+    method: the key is the route placeholder name and the value is the Doctrine
+    property name:
+
+.. code-block:: php
+
+        #[Route('/product/{category}/{slug}/comments/{comment_slug}')]
+        public function show(
+            #[MapDocument(mapping: ['category' => 'category', 'slug' => 'slug'])]
+            Product $product,
+            #[MapDocument(mapping: ['comment_slug' => 'slug'])]
+            Comment $comment
+        ): Response {
+        }
+
+``exclude``
+    Configures the properties that should be used in the ``findOneBy()``
+    method by *excluding* one or more properties so that not *all* are used:
+
+.. code-block:: php
+
+        #[Route('/product/{slug}/{date}')]
+        public function show(
+            #[MapDocument(exclude: ['date'])]
+            Product $product,
+            \DateTime $date
+        ): Response {
+        }
+
+``stripNull``
+    If true, then when ``findOneBy()`` is used, any values that are
+    ``null`` will not be used for the query.
+
+``objectManager``
+    By default, the ``DocumentValueResolver`` uses the *default*
+    object manager, but you can configure this:
+
+.. code-block:: php
+
+        #[Route('/product/{id}')]
+        public function show(
+            #[MapDocument(objectManager: 'foo')]
+            Product $product
+        ): Response {
+        }
+
+``evictCache``
+    If true, forces Doctrine to always fetch the document from the database
+    instead of cache.
```

Comment:
ODM does not provide native cache solutions, I believe we shouldn't be documenting (or exposing) the functionality for now

Reply:
Yes, you're right, I've removed the documentation about this and cannot be configured